microservices/submission-service/src/replay-submission/replay-submission-util.service.ts

```diff
@@ -105,4 +105,88 @@ export class ReplaySubmissionUtilService {
 
         return {canSubmit: true};
     }
+
+    async canRatifySubmission(submissionId: string, playerId: number): Promise<CanRatifySubmissionResponse> {
+        const submission = await this.submissionCrudService.getSubmission(submissionId);
+
+        if (!submission) {
+            return {
+                canRatify: false,
+                reason: "The submission does not exist",
+            };
+        }
+
+        if (!submission.validated) {
+            return {
+                canRatify: false,
+                reason: "The submission has not been validated",
+            };
+        }
+
+        if (submissionIsScrim(submissionId)) {
```

Comment:
submission.type?

Reply:
Meh, we could. This structure matches the `canSubmitReplays` function (these are very similar), so I'm gonna leave it